### PR TITLE
Ews Extension Online Powershell v2 integration

### PR DESCRIPTION
**Pull request by jochman**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/32224

## Description
commands: 
- [x] Get-EXOCASMailbox
- [x] Get-EXOMailbox
- [x] Get-EXOMailboxPermission
- [x] Get-EXORecipient
- [x] Get-EXORecipientPermission

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [x] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.test.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+"""Base Integration for Cortex XSOAR - Unit Tests file
+
+Pytest Unit Tests: all funcion names must start with "test_"
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.test.py:3]: remove

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2_description.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+## BaseIntegration Help
+
+### App authentication
+To use this integration you should connect an application with a certificate.
+In order to create the app and certificate, you should use the [following guide](https://docs.microsoft.com/en-us/powershell/exchange/app-only-auth-powershell-v2?view=exchange-ps).
+
+After acquiring the .pfx certificate, you should encode it to base64 to use in the integration.
+you can use this command:
+```powershell
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2_description.md:9]: update here with how to generate cert

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+  additionalinfo: A pfx certificate encoded in Base64.
+- defaultvalue: ""
+  display: The Organization parameter specifies the organization that's used in app-only authentication
+  name: organization
+  required: false
+  type: 0
+- defaultvalue: ""
+  display: The app_id from the Azure portal
+  name: app_id
+  required: false
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml:24]:
```suggestion
  required: true
```

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+  type: 0
+- display: Certificate
+  name: certificate
+  type: 9
+  required: true
+  additionalinfo: A pfx certificate encoded in Base64.
+- defaultvalue: ""
+  display: The Organization parameter specifies the organization that's used in app-only authentication
+  name: organization
+  required: false
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml:19]:
```suggestion
  required: true
```

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+  type: 0
+- defaultvalue: ""
+  display: The app_id from the Azure portal
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml:22]: improve display name

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+commonfields:
+  id: EWS Extension Online Powershell v2
+  version: -1
+configuration:
+- defaultvalue: https://outlook.office365.com
+  display: Exchange Online URL
+  name: url
+  required: true
+  type: 0
+- display: Certificate
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml:11]: improve error handling for malformed inputs of cert and password

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1 [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+    try {
+        # Creating ExchangeOnline client
+        $exo_client.CreateSession()
+        # Executing command
+        $Demisto.Debug("Command being called is $Command")
+        switch ($command) {
+            "test-module" {
+                ($human_readable, $entry_context, $raw_response) = TestModuleCommand
+            }
+            "$script:COMMAND_PREFIX-get-cas-mailbox" {
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1:354]: move action to end of cmd

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1 [hunk 8]
```diff
@@ -0,0 +1,402 @@
...
+    $identity = $kwargs.identity
+    $organizational_unit = $kwargs.organizational_unit
+    $primary_smtp_address = $kwargs.primary_smtp_address
+    $user_principal_name = $kwargs.user_principal_name
+    $limit = $kwargs.limit -as [int]
+    $raw_response = $client.GetEXOCASMailbox(
+            $identity, $organizational_unit, $primary_smtp_address, $user_principal_name, $limit
+    )
+    $human_readable = TableToMarkdown $raw_response "Results of $command"
+    $entry_context = @{"$script:INTEGRATION_ENTRY_CONTEXT.EXOCASMailBox(obj.Guid === val.Guid)" = $raw_response }
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1:318]:
```suggestion
    $entry_context = @{"$script:INTEGRATION_ENTRY_CONTEXT.CASMailBox(obj.Guid === val.Guid)" = $raw_response }
```

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml [hunk 9]
```diff
@@ -0,0 +1,358 @@
...
+      required: false
+      secret: false
+    - default: false
+      description: The OrganizationalUnit parameter filters the results based on the object's location in Active Directory.
+      isArray: false
+      name: organizational_unit
+      required: false
+      secret: false
+    - default: false
+      description: The PrimarySmtpAddress identifies the mailbox that you want to view by primary SMTP email address. Can't be used with user_principal_name.
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml:45]: please add how to retrieve

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml [hunk 10]
```diff
@@ -0,0 +1,358 @@
...
+      isArray: false
+      name: primary_smtp_address
+      required: false
+      secret: false
+    - default: false
+      description: The UserPrincipalName parameter identifies the mailbox that you want to view by UPN. Can't be used with primary_smtp_address.
+      isArray: false
+      name: user_principal_name
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml:53]: please add how to retrieve

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml [hunk 11]
```diff
@@ -0,0 +1,358 @@
...
+  - arguments:
+    - default: false
+      isArray: false
+      name: identity
+      description: The Identity parameter specifies the mailbox you want to view.
+      required: false
+      secret: false
+    - default: false
+      description: The OrganizationalUnit parameter filters the results based on the object's location in Active Directory.
+      isArray: false
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml:40]: please add how to retrieve

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1 [hunk 12]
```diff
@@ -0,0 +1,402 @@
...
+}
+function GetEXOMailBoxPermissionCommand {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory)][ExchangeOnlinePowershellV2Client]$client,
+        [hashtable]$kwargs
+    )
+    $identity = $kwargs.identity
+    $raw_response = $client.GetEXOMailBoxPermission($identity)
+    $human_readable = TableToMarkdown $raw_response "Results of $command"
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1:280]: convert arraylist to comma seperated str

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1 [hunk 13]
```diff
@@ -0,0 +1,402 @@
...
+
+function GetEXORecipientPermissionCommand {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory)][ExchangeOnlinePowershellV2Client]$client,
+        [hashtable]$kwargs
+    )
+    $identity = $kwargs.identity
+    $limit = $kwargs.limit -as [int]
+    $raw_response = $client.GetEXORecipientPermission($identity, $limit)
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1:267]: please handle Access Right to parse properly in the human readable output

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml [hunk 14]
```diff
@@ -0,0 +1,358 @@
...
+    execution: false
+    name: ews-get-recipient
+  isfetch: false
+  runonce: false
+  script: '-'
+  type: powershell
+  dockerimage: demisto/pwsh-exchange:1.0.0.15724
+fromversion: 5.5.0
+tests:
+- No tests (auto formatted)
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.yml:358]: please add test playbook

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1 (hunk 12 above)
Comment (jochman) [reply Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1:285]: https://github.com/demisto/content/pull/11314

File: Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1 (hunk 13 above)
Comment (jochman) [reply Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1:272]: https://github.com/demisto/content/pull/11314

**Review by hod-alpert — APPROVED** [review]
- [x] Make the certificate secret

**Comment by ShirleyDenkberg** [comment]
@Itay4 @hod-alpert Doc review completed.  READMEs need to be generated for the integration and script.

**Pull request merged**